amd_openvx/openvx/ago/ago_haf_cpu_generic_functions.cpp

```diff
@@ -0,0 +1,878 @@
+                    ((vx_uint32*)destination)[dest_index] = *(vx_uint32*)(ptr + y*stride_y + x*stride_x);
+                    break;
+                default:
+                    abort();
+                }
+            }
+        }
+    }
+    else if (borders->mode == VX_BORDER_CONSTANT)
+    {
+        vx_pixel_value_t cval = borders->constant_value;
+        for (ky = -(int32_t)radius_y; ky <= (int32_t)radius_y; ++ky)
+        {
+            vx_int32 y = (vx_int32)(center_y + ky);
+            int ccase_y = y < 0 || y >= height;
+
+            for (kx = -(int32_t)radius_x; kx <= (int32_t)radius_x; ++kx, ++dest_index)
+            {
+                vx_int32 x = (int32_t)(center_x + kx);
+                int ccase = ccase_y || x < 0 || x >= width;
+
+                switch (type)
+                {
+                case VX_DF_IMAGE_U8:
+                    if (!ccase)
+                        ((vx_uint8*)destination)[dest_index] = *(vx_uint8*)(ptr + y*stride_y + x*stride_x);
+                    else
+                        ((vx_uint8*)destination)[dest_index] = (vx_uint8)cval.U8;
+                    break;
+                case VX_DF_IMAGE_S16:
+                case VX_DF_IMAGE_U16:
+                    if (!ccase)
+                        ((vx_uint16*)destination)[dest_index] = *(vx_uint16*)(ptr + y*stride_y + x*stride_x);
+                    else
+                        ((vx_uint16*)destination)[dest_index] = (vx_uint16)cval.U16;
+                    break;
+                case VX_DF_IMAGE_S32:
+                case VX_DF_IMAGE_U32:
+                    if (!ccase)
+                        ((vx_uint32*)destination)[dest_index] = *(vx_uint32*)(ptr + y*stride_y + x*stride_x);
+                    else
+                        ((vx_uint32*)destination)[dest_index] = (vx_uint32)cval.U32;
+                    break;
+                default:
+                    abort();
+                }
+            }
+        }
+    }
+    else
+        abort();
+}
+
+#define CONV_DIM 5
+#define CONV_DIM_HALF CONV_DIM / 2
+
+#define INSERT_ZERO_Y(slice, y) for (int i=0; i<CONV_DIM; i++) slice[CONV_DIM*(1-y)+i] = 0;
+#define INSERT_VALUES_Y(slice, y) for (int i=0; i<CONV_DIM; i++) slice[CONV_DIM*(high_y-y)+i+CONV_DIM_HALF*CONV_DIM] = slice[CONV_DIM*(high_y-y)+i];
+#define INSERT_ZERO_X(slice, x) for (int i=0; i<CONV_DIM; i++) slice[CONV_DIM*i+1-x] = 0;
+#define INSERT_VALUES_X(slice, x) for (int i=0; i<CONV_DIM; i++) slice[CONV_DIM*i+(high_x-x)+CONV_DIM_HALF] = slice[CONV_DIM*i+(high_x-x)];
+
+static vx_status replicateConvolve(vx_image src, vx_convolution conv, vx_image dst, vx_border_t *bordermode)
+{
+    vx_int32 y, x, i;
+    void *src_base = NULL;
+    void *dst_base = NULL;
+    vx_imagepatch_addressing_t src_addr, dst_addr;
+    vx_rectangle_t rect;
+
+    vx_size conv_width, conv_height;
+    vx_int32 conv_radius_x, conv_radius_y;
+    vx_int16 conv_mat[C_MAX_CONVOLUTION_DIM * C_MAX_CONVOLUTION_DIM] = { 0 };
+    vx_int32 sum = 0, value = 0;
```

Comment:
Codacy found an issue: [Variable 'sum' is assigned a value that is never used.](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pullRequest?prid=5686893)